framework/src/problems/FEProblemBase.C

```diff
@@ -6758,6 +6779,8 @@ FEProblemBase::computeBounds(NonlinearImplicitSystem & libmesh_dbg_var(sys),
   _lower.swap(lower);
   _upper.swap(upper);
 
+  // BWS: In all other cases where this is called, it's within a try/catch to strip
+  // out the exception. Should that be done here too?
   checkExceptionAndStopSolve();
```

Comment:
Yes we should

Reply:
Actually no I don't think we should. I am not 100% confident that this function is called during a solve, in which case I would rather have an uncaught exception that leads to simulation termination than a silenced exception

Reply:
OK, I'll just remove that comment and consider that resolved.

I do wonder whether it's worthwhile to add a parameter to `checkExceptionAndStopSolve()` (or another variant of it) that makes it not throw an exception. There are an awful lot of try/catch blocks around calls to this just to strip out the exception.

Reply:
Ok I looked into this more. This is called during setup of the nonlinear solver, so not during the Newton iterations. So if we have an exception at this point, then it is like having an exception for execution flags `!EXEC_LINEAR && !EXEC_NONLINEAR`, in which case we should mooseError. So I think in this function we should wrap the `_aux->compute` in a `try` and then in the `catch` we should `mooseError`